context.go

```diff
@@ -67,6 +65,65 @@ type SpanContext struct {
 	//
 	// See JaegerDebugHeader in constants.go
 	debugID string
+
+	// samplingState is shared across all spans
+	samplingState *samplingState
+}
+
+type samplingState struct {
+	_flags atomic.Int32 // Only lower 8 bits are used. We use an int32 instead of a byte to use CAS operations
+}
+
+func (s *samplingState) setFlag(newFlag int32) {
+	swapped := false
+	for !swapped {
+		old := s._flags.Load()
+		swapped = s._flags.CAS(old, old|newFlag)
+	}
+}
+
+func (s *samplingState) resetFlag(newFlag int32) {
```

Comment:
"reset" semantically means "return to previous state". I think it's more common to use "unset" for this operation.

Reply:
> "reset" semantically means "return to previous state".

I disagree - I think this definition from Oxford is more apt - "ELECTRONICS cause (a binary device) to enter the state representing the numeral 0."

I've seen `revert` used to mean to "return to previous state" 

That being said I don't have a strong preference between `unset` and `reset` - If you feel strongly I can update to `unset`.

Reply:
That is very obscure meaning. "Reset to factory settings" means change to default state, it doesn't mean to 0.

I think set/unset are a better duals.